### ISSUE TITLE
Rectangle and Diamond produce different corner shapes

### ISSUE THREAD

**Issue by jai-dewani** [excalidraw/excalidraw#4375]
Thanks to @dai-shi for pointing out this issue (Refer the image at the end of description) 

This bug exists because rectangle and diamond corners are build using different curves (Quadrative vs Cubic curves). As discussed with @vjeux in this [twitter thread](https://twitter.com/dai_shi/status/1467818603089215493), the aim to select a single type of corner and use it across different shapes. 

![image](https://user-images.githubusercontent.com/33520257/144886521-d0295b6b-ac1f-4b0f-81a5-9011fcf0476a.png)

**Comment by dwelle** [comment]
I did notice, but didn't think it worth mentioning. At some point my idea is to merge the two implementations completely. In fact, I envision making all shapes into lines, possibly just with a couple different rules for editing (or disallowing editing).

**Comment by dai-shi** [comment]
I thought it's nice a hack for now. Technically, C-curves can replicate Q-curves, right? I just picked Q, which is less debatable.

**Comment by dwelle** [comment]
![image](https://user-images.githubusercontent.com/5153846/145041237-7a818881-6338-4f81-9a60-fbde9f9bab71.png)

If we're gonna align, I actually like C-curves better at bigger sizes because the roundness is more subtle (i.e. it scales better). Some users pointed this out, and it seems you agree as well @dai-shi https://github.com/excalidraw/excalidraw/issues/3600

**Comment by dai-shi** [comment]
I agree as a user, if the box is very big, it feels too round. But, it really depends.
Also, with a little bit of playing around, it might be interesting add Math.min (or max) for diamonds too for aspect ratio not close to 1.

All that said, I don't personally think the aligning is so crucial, and it might incur some other difficulties.

> In fact, I envision making all shapes into lines, possibly just with a couple different rules for editing (or disallowing editing).

I like the idea.

**Comment by jai-dewani** [comment]
>  it might be interesting add Math.min (or max) for diamonds too for aspect ratio not close to 1.

What do you mean by adding Math.min (or max)? A max-min limit for the roundness of the corners?

**Comment by dai-shi** [comment]
Sorry for the confusion. I meant this `Math.min`:
https://github.com/excalidraw/excalidraw/blob/cecabc2196252c338cfbd5a77ed0ad8ff05f758c/src/renderer/renderElement.ts#L381

But, it may not make sense for diamonds. It does for rectangles.

**Comment by dixudx** [comment]
Any progress on this wonderful feature? It is quite handy. Currently the roundness is fixed, which is devoid of beauty for a large rectangle.

**Comment by dwelle** [comment]
@dixudx yup, but we're working on making the edges align more closely across different sizes: https://github.com/excalidraw/excalidraw/pull/5553 — afterwards, we'll see if customization is really needed!

**Comment by jai-dewani** [comment]
After the #5553 merge, the difference between them has been reduced drastically!

The green one is Diamond and the red is Rectangle 

![image](https://user-images.githubusercontent.com/33520257/207235003-4ee5a1c9-e271-479b-a5ea-6dfe6b050de4.png)

When I try to overlap them together 

![image](https://user-images.githubusercontent.com/33520257/207235183-42012e00-72b9-4171-bd30-e4d32394dfeb.png)

With architect slope for a clear distinction 

![image](https://user-images.githubusercontent.com/33520257/207235208-421c8892-980c-45e5-8e79-ea62293c94ed.png)
